정민지/mvvmstudy/app/src/main/java/com/github/sookhee/mvvmstudy/ui/main/MainActivity.kt

```diff
@@ -0,0 +1,99 @@
+package com.github.sookhee.mvvmstudy.ui.main
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.addTextChangedListener
+import com.github.sookhee.mvvmstudy.databinding.ActivityMainBinding
+import com.github.sookhee.mvvmstudy.interactor.MainInteractor
+import com.github.sookhee.mvvmstudy.model.GithubRepositoryModel
+import com.github.sookhee.mvvmstudy.ui.detail.DetailActivity
+
+/**
+ *  MainActivity.kt
+ *
+ *  Created by Minji Jeong on 2022/04/25
+ *  Copyright © 2022 MashUp All rights reserved.
+ */
+
+class MainActivity : AppCompatActivity(), MainContract.View {
+    private lateinit var binding: ActivityMainBinding
+
+    private val interactor by lazy { MainInteractor() }
+    private val presenter by lazy { MainPresenter(this, interactor) }
+    private val repositoryAdapter by lazy {
+        RepositoryAdapter().apply {
+            onItemClick = {
+                val intent = Intent(this@MainActivity, DetailActivity::class.java)
+                intent.putExtra(EXTRA_REPOSITORY_ID, it.id)
+
+                startActivity(intent)
```

Comment:
너무 많은 중첩이긴 하지만.. ㅋㅋㅋㅋㅋ 한번더 apply로 묶을 수도 있겠네요!
참고만 해주세요!

```suggestion
                val intent = Intent(this@MainActivity, DetailActivity::class.java)
                intent.putExtra(EXTRA_REPOSITORY_ID, it.id)

                startActivity(
                    Intent(this@MainActivity, DetailActivity::class.java).apply {
                           putExtra(EXTRA_REPOSITORY_ID, it.id)
                    }
                )
```